### PR TITLE
[SCB-1532] Redis configuration is invalid

### PR DESCRIPTION
**Pull request by Fxdemon**
```
#redis 单机模式
#akkaConfig:
#  akka.persistence.journal.plugin: akka-persistence-redis.journal
#  akka.persistence.snapshot-store.plugin: akka-persistence-redis.snapshot
#  akka-persistence-redis:
#    redis:
#      mode: simple
#      host: ${spring.redis.host}
#      port: ${spring.redis.port}
#      database: 0
#      #password: ${spring.redis.password}

#redis 哨兵模式
akkaConfig:
  akka.persistence.journal.plugin: akka-persistence-redis.journal
  akka.persistence.snapshot-store.plugin: akka-persistence-redis.snapshot
  akka-persistence-redis:
    redis:
      mode: sentinel
      database: 1
      password: ${spring.redis.password}
      master: ${spring.redis.sentinel.master}
```
      sentinel-list: ${spring.redis.sentinel.nodes}

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by coolbeevip** [comment]
直接在 application.yaml 中增加配置不行吗？

**Comment by Fxdemon** [comment]
不行

**Comment by coolbeevip** [comment]
> 不行

你启动的时候配置了启动集群环境参数了吗？

```
--spring.profiles.active=prd,cluster
```

**Comment by coolbeevip** [comment]
我们希望对于redis的支持可以通过启动参数配置，而不是通过重新编译源代码的方式，按照你修改后的解析方式，我理解应该就支持以下这种方式了，你能确认一下吗？

redis 单例模式

```
--akkaConfig.akka-persistence-redis.redis.mode=simple
--akkaConfig.akka-persistence-redis.redis.host=127.0.0.1
--akkaConfig.akka-persistence-redis.redis.port=6379
--akkaConfig.akka-persistence-redis.redis.database=0
--akkaConfig.akka-persistence-redis.redis.password=xxx
```

redis 哨兵模式

```
--akkaConfig.akka-persistence-redis.redis.mode=sentinel
--akkaConfig.akka-persistence-redis.redis.host=127.0.0.1
--akkaConfig.akka-persistence-redis.redis.port=6379
--akkaConfig.akka-persistence-redis.redis.database=0
--akkaConfig.akka-persistence-redis.redis.master=mymaster
--akkaConfig.akka-persistence-redis.redis.sentinel-list=127.0.0.1:6379
--akkaConfig.akka-persistence-redis.redis.password=xxx
```

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/spring/integration/akka/AkkaConfigPropertyAdapter.java [hunk 1]
```diff
@@ -58,7 +59,18 @@
               if (LOG.isTraceEnabled()) {
                 LOG.trace("Adding property {}={}" + key, value);
               }
+              
               propertyMap.put(key, value);
+              
+              if(name.startsWith(REDIS_NAME) && !propertyMap.containsKey(name)){
+                String readJournalKey = ("akka-persistence-redis.read-journal.redis.").concat(name.substring(REDIS_NAME.length()));
+                String journalKey = ("akka-persistence-redis.journal.redis.").concat(name.substring(REDIS_NAME.length()));
```
Comment (WillemJiang) [inline alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/spring/integration/akka/AkkaConfigPropertyAdapter.java]: why do we need to set up 3 configuration key with one value?

**Comment by Fxdemon** [comment]
> --akkaConfig.akka-persistence-redis.redis.mode=sentinel
> --akkaConfig.akka-persistence-redis.redis.database=0
> --akkaConfig.akka-persistence-redis.redis.master=mymaster
> --akkaConfig.akka-persistence-redis.redis.sentinel-list=127.0.0.1:6379
> --akkaConfig.akka-persistence-redis.redis.password=xxx
如上配akka-persistence-redis.jar  读取不到对应配置信息， 目前默认读取akka-persistence-redis.jar中 *.conf中的配置 

使用方式：
`#redis 单机模式
#akkaConfig:
#  akka.persistence.journal.plugin: akka-persistence-redis.journal
#  akka.persistence.snapshot-store.plugin: akka-persistence-redis.snapshot
#  akka-persistence-redis:
#    redis:
#      mode: simple
#      host: ${spring.redis.host}
#      port: ${spring.redis.port}
#      database: 0
#      #password: ${spring.redis.password}

#redis 哨兵模式
akkaConfig:
  akka.persistence.journal.plugin: akka-persistence-redis.journal
  akka.persistence.snapshot-store.plugin: akka-persistence-redis.snapshot
  akka-persistence-redis:
    redis:
      mode: sentinel
      database: 1
      password: ${spring.redis.password}
      master: ${spring.redis.sentinel.master}
      sentinel-list: ${spring.redis.sentinel.nodes}`

**Comment by coolbeevip** [comment]
> > --akkaConfig.akka-persistence-redis.redis.mode=sentinel
> > --akkaConfig.akka-persistence-redis.redis.database=0
> > --akkaConfig.akka-persistence-redis.redis.master=mymaster
> > --akkaConfig.akka-persistence-redis.redis.sentinel-list=127.0.0.1:6379
> > --akkaConfig.akka-persistence-redis.redis.password=xxx
> > 如上配akka-persistence-redis.jar  读取不到对应配置信息， 目前默认读取akka-persistence-redis.jar中 *.conf中的配置
> 
> 使用方式：
> 
> ```
> #redis 单机模式
> #akkaConfig:
> #  akka.persistence.journal.plugin: akka-persistence-redis.journal
> #  akka.persistence.snapshot-store.plugin: akka-persistence-redis.snapshot
> #  akka-persistence-redis:
> #    redis:
> #      mode: simple
> #      host: ${spring.redis.host}
> #      port: ${spring.redis.port}
> #      database: 0
> #      #password: ${spring.redis.password}
> 
> #redis 哨兵模式
> `akkaConfig:
>   akka.persistence.journal.plugin: akka-persistence-redis.journal
>   akka.persistence.snapshot-store.plugin: akka-persistence-redis.snapshot
>   akka-persistence-redis:
>     redis:
>       mode: sentinel
>       database: 1
>       password: ${spring.redis.password}
>       master: ${spring.redis.sentinel.master}
>       sentinel-list: ${spring.redis.sentinel.nodes}`
> ```
Ok，你能同步更新一下这个文档吗？
https://github.com/apache/servicecomb-pack/blob/master/docs/fsm/fsm_manual_zh.md
https://github.com/apache/servicecomb-pack/blob/master/docs/fsm/fsm_manual.md

File: alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/spring/integration/akka/AkkaConfigPropertyAdapter.java (hunk 1 above)
Comment (coolbeevip) [reply alpha/alpha-fsm/src/main/java/org/apache/servicecomb/pack/alpha/fsm/spring/integration/akka/AkkaConfigPropertyAdapter.java]: > why do we need to set up 3 configuration key with one value?
It's OK, akka-persistence-redis contain three components, journal, snapshot, read-journal, their Redis configuration is the same.

https://github.com/safety-data/akka-persistence-redis/blob/master/src/main/resources/reference.conf

**Pull request merged**